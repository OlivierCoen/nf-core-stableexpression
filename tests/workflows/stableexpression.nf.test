nextflow_workflow {

    name "Test Workflow STABLEEXPRESSION"
    script "workflows/stableexpression.nf"
    workflow "STABLEEXPRESSION"

    test("Two Expression Atlas accessions provided") {

        tag "workflow_eatlas_accessions"

        when {
            params {
                species = 'solanum tuberosum'
                fetch_from_expression_atlas = false
                expression_atlas_accessions = "E-MTAB-552,E-GEOD-61690"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas keywords provided") {

        tag "workflow_eatlas_kw"

        when {
            params {
                species = 'solanum tuberosum'
                fetch_expression_atlas_accessions = true
                expression_atlas_keywords = "potato,stress"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas keywords provided - Normalization with EdgeR") {

        tag "workflow_eatlas_kw_edger"

        when {
            params {
                species = 'solanum tuberosum'
                fetch_expression_atlas_accessions = true
                expression_atlas_keywords = "potato,stress"
                normalization_method = "edger"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Input dataset provided") {

        tag "workflow_datasets"

        when {
            params {
                datasets = "tests/input/custom_datasets/input.csv"
                species = 'solanum tuberosum'
                fetch_expression_atlas_accessions = false
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Full test") {

        tag "workflow_full"

        when {
            params {
                species = 'solanum tuberosum'
                datasets = "tests/input/custom_datasets/input.csv"
                expression_atlas_accessions = "E-MTAB-552,E-GEOD-61690"
                fetch_expression_atlas_accessions = true
                expression_atlas_keywords = "phloem"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

}
