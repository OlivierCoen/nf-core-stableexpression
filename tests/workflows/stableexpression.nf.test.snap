{
    "Two Expression Atlas keywords provided": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,bbf85cf21180ab7306d23f3fac07ea14"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,bbf85cf21180ab7306d23f3fac07ea14"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-28T23:44:22.027225612"
    }
}